### PR TITLE
chore(ICRC-Rosetta): FI-1401: add secp key test

### PR DESCRIPTION
**Pull request by NikolasHai**
This MR introduces the following changes:

1. Add support for Secp256k1 keys to Rosetta Client
2. Modify one of the system tests in ICRC Rosetta to also use Secp256k1 keys to make sure the signing through the rosetta client works as expected.

### CODE REVIEW

File: rs/rosetta-api/icrc1/rosetta/client/src/lib.rs [hunk 1]
```diff
@@ -69,22 +70,46 @@ impl RosettaClient {
...
+                        .verify_signature(signable_bytes.as_slice(), signed_bytes.as_slice())
+                    {
+                        bail!("Signature verification failed")
+                    };
+                }
+                _ => bail!(
+                    "Unsupported curve type: {:?}",
+                    signer_keypair.get_curve_type()
+                ),
+            }
```
Comment (mbjorkqvist) [inline rs/rosetta-api/icrc1/rosetta/client/src/lib.rs:111]: I was going to suggest adding some tests for these error cases, but I couldn't think of how you could reach them?

**Review by mbjorkqvist — APPROVED** [review]
There's a number of rosetta blocks changes showing up in the diff, but I would imagine they'd go away when merging in master once more? Otherwise LGTM!

File: rs/rosetta-api/icrc1/rosetta/client/src/lib.rs (hunk 1 above)
Comment (NikolasHai) [reply rs/rosetta-api/icrc1/rosetta/client/src/lib.rs:111]: The rosetta client is only for our use really, so would it be safe to say we are only interested in it working as intended for testing?

Comment (mbjorkqvist) [reply rs/rosetta-api/icrc1/rosetta/client/src/lib.rs:111]: Ah, I see, thanks for the clarification!

**Pull request merged**